samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java

```diff
@@ -1466,6 +1468,36 @@ private double getScoreByQuestionType(ItemGradingData itemGrading, ItemDataIfc i
     return autoScore;
   }
 
+  private boolean isThisItemDistractor(ItemDataIfc item, ItemGradingData thisItemGradingData){
+    Iterator<ItemTextIfc> itemIter = item.getItemTextArraySorted().iterator();
+    while (itemIter.hasNext()) {
+      ItemTextIfc curItem = itemIter.next();
+      if (isDistractor(curItem)) {
+        if (curItem.getId().doubleValue() == thisItemGradingData.getPublishedItemTextId().doubleValue()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  private double getScoreDistractor(ItemDataIfc item, ItemGradingData thisItemGradingData){
+    double numItems = 0;
```

Comment:
Looks like this is just a duplication of the numberOfChoices logic above. I think this method could just return the "initScore" the same way getAnswerScore() does and then we don't need the duplication. Basically change to something like

```
initScore = isDistractor? getDistractorScore() : getAnswerScore();
```